designsafe/static/scripts/ng-designsafe/components/modals/data-browser-service-preview/data-browser-service-preview.component.js

```diff
@@ -69,6 +70,11 @@ class DataBrowserServicePreviewCtrl {
                     };
                     oReq.send();
                 }
+                if (this.fileType === 'hazmapper') {
```

Comment:
I have a suggestion for a change in functionality here. If we parse the ID out of the filename then renaming the file will break the link. I think that a better strategy would be to put the hazmapper ID in the body of the file, where we can retrieve it using similar code to `if (this.fileType === 'text') {...`. That way we can have `.hazmapper` files with helpful names that still point to the correct maps.

Reply:
@jarosenb Changed to get information from the file body.